src/fsharp/FSharp.Core/array.fs

```diff
@@ -559,6 +559,31 @@ namespace Microsoft.FSharp.Collections
                     res.[i] <- Microsoft.FSharp.Primitives.Basics.Array.subUnchecked i windowSize array
                 res
 
+        [<CompiledName("ChunksOf")>]
+        let chunksOf chunkSize (array:'T[]) =
+            checkNonNull "array" array
+            if chunkSize <= 0 then invalidArg "chunkSize" (SR.GetString(SR.inputMustBeNonNegative))
+            let len = array.Length
+            if len = 0 then
+                [| |]
+            else if chunkSize > len then
+                [| array |]
```

Comment:
We should align with established pattern of always returning fresh data, rather than including pointer pack to input array.